sys/log/full/src/log_shell.c

```diff
@@ -45,49 +167,70 @@ shell_log_dump_entry(struct log *log, struct log_offset *log_offset,
     int dlen;
     int rc;
 #if MYNEWT_VAL(LOG_VERSION) > 2
+    struct CborParser cbor_parser;
+    struct CborValue cbor_value;
+    struct log_shell_cbor_reader cbor_reader;
     char tmp[32 + 1];
     int off;
     int blksz;
+    bool read_data = ueh->ue_etype != LOG_ETYPE_CBOR;
+#else
+    bool read_data = true;
 #endif
 
     dlen = min(len, 128);
 
-    rc = log_read_body(log, dptr, data, 0, dlen);
-    if (rc < 0) {
-        return rc;
+    if (read_data) {
+        rc = log_read_body(log, dptr, data, 0, dlen);
+        if (rc < 0) {
+            return rc;
+        }
+        data[rc] = 0;
     }
-    data[rc] = 0;
+
+    console_printf("[%llu] ", ueh->ue_ts);
 
 #if MYNEWT_VAL(LOG_VERSION) <= 2
-    console_printf("[%llu] %s\n", ueh->ue_ts, data);
+    console_write(data, strlen(data));
 #else
     switch (ueh->ue_etype) {
     case LOG_ETYPE_STRING:
-        console_printf("[%llu] %s\n", ueh->ue_ts, data);
+        console_write(data, strlen(data));
+        break;
+    case LOG_ETYPE_CBOR:
+        log_shell_cbor_reader_init(&cbor_reader, log, dptr, len);
+        cbor_parser_init(&cbor_reader.r, 0, &cbor_parser, &cbor_value);
+        cbor_value_to_pretty(stdout, &cbor_value);
```

Comment:
printing to stdout would end up on console as other output, right?

Reply:
correct